low-level/framework/messaging/egress_processor.py

```diff
@@ -205,16 +211,22 @@ def _transmit_msg_on_exchange(self):
                          "actuator_response_type").get(
                          "thread_controller") is not None):
                 self._add_signature()
-                self._producer.send([json.dumps(self._jsonMsg)])
+                if isinstance(self._producer, MessageProducer):
+                    self._producer.send([json.dumps(self._jsonMsg)])
+                else:
+                    self.create_MsgProducer_obj()
                 logger.debug(
                     "_transmit_msg_on_exchange, Successfully Sent: %s" % self._jsonMsg)
             else:
                 self._add_signature()
                 jsonMsg = json.dumps(self._jsonMsg)
                 try:
                     if self.store_queue.is_empty():
-                        self._producer.send([jsonMsg])
-                        logger.info(f"Published Alert: {jsonMsg}")
+                        if isinstance(self._producer, MessageProducer):
+                            self._producer.send([jsonMsg])
+                            logger.info(f"Published Alert: {jsonMsg}")
+                        else:
+                            self.create_MsgProducer_obj()
```

Comment:
Same quesiton, If this fails or pass, what happens to current message being processed ?

Reply:
if msg_producer instance is not available then in 'else' part, adding that msg in accumulated queue, sit will be processed by accumulated processor.